### PR TITLE
[Model] Support nested structures for TensorSchema

### PR DESCRIPTION
**Pull request by DarkLight1337**
## Purpose

This helps simplify the implementation of HyperCLOVAX-SEED-Vision and enable `merge_by_field_config=True` for that model.

Other changes:

- Improve the name of the field in `Phi3VImagePixelInputs`
- Fix invalid import for GLM4.1V caused by #25631

cc @bbeckca 

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/model_executor/models/hyperclovax_vision.py [hunk 1]
```diff
@@ -246,51 +229,43 @@ def replace_multimodal_token(
...
-                    for _i in range(len(videos))
-                ]
-                _processed_outputs["vision_query_lengths_videos"] = [
-                    _processed_outputs["vision_query_lengths_videos"]
-                    [sum(_num_per_videos[:_i]):sum(_num_per_videos[:_i + 1])]
-                    for _i in range(len(videos))
+                    torch.stack(_processed_outputs["pixel_values_videos"]
+                                [_idx_per_video[i]:_idx_per_video[i + 1]])
+                    for i in range(len(videos))
                 ]
```
[the remark this answers is not in the document]
Comment (DarkLight1337) [reply vllm/model_executor/models/hyperclovax_vision.py:248]: Fixing

File: vllm/model_executor/models/hyperclovax_vision.py [hunk 2]
```diff
@@ -246,51 +229,43 @@ def replace_multimodal_token(
...
-                        ],
-                    ) for _input_ids in processed_outputs["input_ids"]
-                ],
-                                                             dim=0)
+                _processed_outputs[
+                    "vision_query_lengths_videos"] = torch.tensor([
+                        _processed_outputs["vision_query_lengths_videos"]
+                        [_idx_per_video[i]:_idx_per_video[i + 1]]
+                        for i in range(len(videos))
+                    ])
```
[the remark this answers is not in the document]
Comment (DarkLight1337) [reply vllm/model_executor/models/hyperclovax_vision.py]: Fixing

File: vllm/model_executor/models/hyperclovax_vision.py [hunk 3]
```diff
@@ -692,55 +653,94 @@ def get_placeholder_str(cls, modality: str, i: int) -> Optional[str]:
...
+            if (input_key == "pixel_values_images"
+                    and "images" not in modalities):
+                modalities["images"] = self._parse_and_validate_image_input(
+                    **kwargs)
+            if (input_key == "pixel_values_videos"
+                    and "videos" not in modalities):
+                modalities["videos"] = self._parse_and_validate_video_input(
+                    **kwargs)
+
+        return modalities
```
[the remark this answers is not in the document]
Comment (DarkLight1337) [reply vllm/model_executor/models/hyperclovax_vision.py:715]: `**kwargs` is unpacked into the method so this doesn't happen

**Review by Isotr0py — APPROVED** [review]
LGTM!
